tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_multi_mount.ksh

```diff
@@ -0,0 +1,99 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy is of the CDDL is also available via the Internet
+# at http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright(c) 2018 Datto Inc.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify multi mount functionality
+#
+# STRATEGY:
+# 1. Create fs
+# 2. Create and hold open file in filesystem
+# 3. Lazy unmount
+# 4. Verify remounting fs that was lazily unmounted is possible
+# 5. Verify multiple mounts of the same dataset are possible
+# 6. Verify bind mount doesn't prevent rename
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	datasetexists $TESTDS && datasetdestroy $TESTDS
```

Comment:
There is no `datasetdestroy`  funciton.  You can use the `destroy_dataset` hlper function for this.  You should also make sure to explicitly unmount the additional mountpoints first.

```
        destroy_dataset "$TESTDS" "-f"
```

Reply:
Damn it, I left out `log_must` on that destroy so didn't notice that function didn't exist... Made updates to cleanup and it's looking better now:
```NOTE: Performing local cleanup via log_onexit (cleanup)
SUCCESS: umount /testpool/multi-mount-test
SUCCESS: umount /testpool/multi-mount-test-second
SUCCESS: umount /testpool/multi-mount-test-third
SUCCESS: zfs destroy -f testpool/multi-mount-test
SUCCESS: destroy_dataset testpool/multi-mount-test -f```

Reply:
Much better!  Thanks, just waiting on the test results now.